pkg/util/retry/retry.go

```diff
@@ -34,10 +35,49 @@ func Do(ctx context.Context, fn func() error, opts ...Option) error {
 		opt(c)
 	}
 
-	var el error
+	if c.totalTimeOut != 0 {
+		var el error
```

Comment:
all the go timeout should be controlled by context but not through a time check.
The timeout may has following issues:
1) not accurate
2) if timeouted but connect succeed finally, there maybe resource leakage because no one is gonna to handle timeouted request